car_service/request_service.rb

```diff
@@ -0,0 +1,118 @@
+require_relative 'request_repo'
+
+#computed in seconds
+module TimeUtils
+  HOUR = 3600
+  DAY = 24 * 3600
+  MINUTE = 60
+end
+
+#schedule of the car wash
+module Schedule
+  OPENING_TIME = 8
+  CLOSING_TIME = 18
+  OPENING_TIME_WEEKEND = 9
+  CLOSING_TIME_WEEKEND = 15
+end
+
+class RequestService
+  include Schedule
+  include TimeUtils
+
+  def initialize
+    #one for each worker, 2 cars can be washed at the same time
+    @schedule1 = Repo.new
+    @schedule2 = Repo.new
+  end
+
+  # @param [Time] time
+  def add_request
+    new_request = Request.new Time.now
+    set_pickup_date new_request
+    pickup_date_print new_request
+  end
+
+  #pretty print of the time
+  def pickup_date_print(request)
+    request.pickup_time.strftime("%A %d-%m-%Y %H:%M")
+  end
+
+  private
+  # @param [Request] request
+  #computes and sets the pickup time attribute of the request
+  def set_pickup_date(request)
+    #check to see which worker will wash the car and set the starting time for the washing
+    # check if the arrays are empty
+    if @schedule1.is_empty?
+      request.wash_time = request.arrive_time
+      compute_pickup_date request
+      @schedule1.add_request request
+      return
+    elsif @schedule2.is_empty?
+      request.wash_time = request.arrive_time
+      compute_pickup_date request
```

Comment:
those lines are duplicate. You could make them before the if, and only assign it to the. correct schedule inside the if statement